push_to_docker.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# The image build step occurs only when a build is triggered by a commit being merged into the "master" branch.
+# Given that the "master" branch is protected, the only way to trigger a build from the "master" branch is when a PR
+# is merged into it. This way, we avoid storing images from non-stable branches.
+VERSION=$(cat ./version/version.go | grep "var VERSION" | awk ' { print $4 } ' | sed s/\"//g)
+VERSION_REGEX="^$VERSION$"
+TAG=$(curl https://registry.hub.docker.com/v1/repositories/tfgco/khan/tags | \
+    sed -e 's/[][]//g' -e 's/"//g' -e 's/ //g' | \
+    tr '}' '\n'  | \
+    awk -F: '{print $3}' | \
+    grep "$VERSION_REGEX"
+)
+
+if [ "$TAG" = "$VERSION" ]; then
+  echo "An image with this version already exists in Docker Hub. Please update your version.go file and try again."
+fi
+
+docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
+
+docker build -t will-iam .
+docker tag will-iam:latest tfgco/will-iam:$VERSION
+docker tag will-iam:latest tfgco/will-iam:latest
```

Comment:
Let's not use `latest`, it's poor docker practice. I'd much rather have `"tfgco/will-iam:$commit_hash"`

Reply:
Completely forgot to add the SHA based tag, gonna add it right away 👍 . As for the `latest` tag, I understand that it may be "dangerous", but it provides some convenience and I'm not 100% sure if we should abandon it 🤔 💭

Reply:
I'd definitely remove latest, IMO we don't want that tag to exist. (I'd remove it from dockerhub too.)

Reply:
If we don't update it here, we should definitely remove it from DockerHub – having a `:latest` tag that _isn't_ the latest seems like an even worse antipattern than simply having a `:latest` tag.

I had a look around and the rest of the internet seems to vary between "be wary of :latest, it doesn't actually mean latest" and "never use :latest, package owners should remove it immediately". Accordingly, if we get specific workflows that want `:latest`, I think it makes sense to consider it (so long as these workflows are careful), but since it looks like [this DockerHub](https://hub.docker.com/r/tfgco/will-iam/tags) is new, maybe we should avoid it until/unless it's needed?

Reply:
Absolutely, we don't want to have a broken `latest` tag 👍

Reply:
I ended up removing the `:latest` until we actually need it. Better safe than sorry won this time 😄